superset/db_engine_specs/presto.py

```diff
@@ -509,13 +507,13 @@ def where_latest_partition(  # pylint: disable=too-many-arguments
         }
 
         for col_name, value in zip(col_names, values):
-            if col_name in column_type_by_name:
-                if column_type_by_name.get(col_name) == "TIMESTAMP":
-                    query = query.where(Column(col_name, TimeStamp()) == value)
-                elif column_type_by_name.get(col_name) == "DATE":
-                    query = query.where(Column(col_name, Date()) == value)
-                else:
-                    query = query.where(Column(col_name) == value)
+            col_type = column_type_by_name.get(col_name)
+
+            if isinstance(col_type, (types.DATE, types.TIMESTAMP)):
+                value = text(f"{col_type} '{value}'")
```

Comment:
I'd ideally like to rely on sqlalchemy to correctly format my value, but I couldn't find a way to do it. Attempting to use `cast()` didn't work, for example, as sqlalchemy told me it didn't know how to literally print the string we have here. I couldn't see a trivial way of constructing the relevant sqla type from the string value we have, either.

Reply:
Actually when I went looking for the dialect, I found `presto_sql_types` which seem to be the correct types to use to get the `DATE '...'` functionality, so replacing it with this perhaps does the job better (I have just tested this, though I still need to fix my unit test to deal with tokenisation for it to work there):

```python
            if isinstance(col_type, types.DATE):
                col_type = Date()  # from superset.models.sql_types.presto_sql_types
            elif isinstance(col_type, types.TIMESTAMP):
                col_type = TimeStamp()

            query = query.where(Column(col_name, col_type) == value)
```

Reply:
Updated to use these types and adjusted tests.